### PR TITLE
doc: add performance.md

### PR DESCRIPTION
**Pull request by xiang90**
/cc @heyitsanthony @joshix @gyuho

### CODE REVIEW

File: Documentation/op-guide/performance.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+There are other sub-systems which impact the overall performance of etcd. Each serialized etcd request must run through etcd’s boltdb-backed MVCC storage engine, which usually takes tens of microseconds to finish. Periodically etcd incrementally snapshots its recently applied requests, merging them back with the previous on-disk snapshot. This process may lead to a latency spike. Although this is usually not a problem on SSDs, it may double the observed latency on HDD. Likewise, inflight compactions can impact etcd’s performance. Fortunately, the impact is often insignificant since the compaction is staggered so it does not compete for resources with regular requests. The RPC system, gRPC, gives etcd a well-defined, extensible API, but it also introduces additional latency, especially for local reads.
```
Comment (joshix) [inline Documentation/op-guide/performance.md]: s/sub-systems/subsystems/

File: Documentation/op-guide/performance.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# Performance
+
+## Understanding performance
+
+etcd provides stable, sustained high performance. Two factors define performance: latency and throughput. Latency is the time taken to complete an operation. Throughput is the total operations completed within some time period. Usually average latency increases as the overall throughput increases when etcd accepts concurrent client requests. On a commodity cloud environment (e.g., standard n-4 on GCE or a comparable machine type on AWS), a three member etcd cluster finishes a request in less than one millisecond under light load, and can complete more than 30,000 requests per second under heavy load.
```
Comment (joshix) [inline Documentation/op-guide/performance.md]: "In common cloud environments, like a standard `n-4` on Google Compute Engine (GCE) or a comparable machine type on AWS, ..."

File: Documentation/op-guide/performance.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+etcd uses the Raft consensus algorithm to replicate requests among members and reach agreement. Consensus performance, especially commit latency, is limited by two physical constraints: network IO latency and disk IO latency. The minimum time to finish an etcd request is the network round-trip-time between members plus fdatasync time. The RTT within  a datacenter may be as low as several hundred microseconds. A typical RTT within the United States is around 50ms and can be as slow as 400ms across continents. The typical fdatasync latency for spinning disk is about 10ms. For SSDs, the latency is often lower than 1ms. To increase throughput, etcd batches multiple requests together and submit them to Raft. This batching policy lets etcd attain high throughput despite heavy load.
```
Comment (joshix) [inline Documentation/op-guide/performance.md]:
```
s/round-trip-time/Round Trip Time (RTT)/
s/between members/between members, plus/
s/fdatasync/the time `fdatasync`requires to commit the data to permanant storage./
s/within  a/within a/
s/50ms/50ms,/
s/slow/long/
s/across/between/
s/spinning/a spinning/
s/submit/submits/
```

**Comment by joshix** [comment]
reviewed, suggestions inline.

**Comment by xiang90** [comment]
@joshix All fixed. PTAL.

**Comment by gyuho** [comment]
LGTM. Thanks.

**Comment by heyitsanthony** [comment]
lgtm

**Pull request merged**